src/components/RegisterForm/index.tsx

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Button, TextField } from '@mui/material'
+import { useFormik } from 'formik'
+import { useContext } from 'react'
+import { useMutation } from 'react-relay'
+import { useNavigate } from 'react-router-dom'
+import { toast } from 'react-toastify'
+import { AuthContext } from '../../context/AuthContext'
+import { UserCreate } from '../../mutations/userCreateMutation'
+import Heading from '../Heading'
+import { Form, FormContainer } from './style'
+
+export const FormRegister = () => {
+  const { login } = useContext(AuthContext)
+  const [loginUser] = useMutation(UserCreate)
+  const navigate = useNavigate()
+  const formikValue = useFormik({
+    initialValues: {
+      username: '',
+      email: '',
+      password: ''
+    },
+    onSubmit: (values, actions) => {
+      loginUser({
+        variables: {
+          input: {
+            username: values.username,
+            email: values.email,
+            password: values.password
+          }
+        },
+        onCompleted({ userCreate }: any) {
+          if (!userCreate.error) {
+            const token = userCreate.token
+            login(token)
+            navigate('/')
+            toast.success('Logged in!', {
+              position: toast.POSITION.TOP_RIGHT,
+              autoClose: 2000,
+              hideProgressBar: false,
+              closeOnClick: true,
+              pauseOnHover: true,
+              draggable: true,
+              progress: undefined
+            })
+          } else {
+            toast.error('This user already exists!', {
+              position: toast.POSITION.TOP_RIGHT,
+              autoClose: 2000,
+              hideProgressBar: false,
+              closeOnClick: true,
+              pauseOnHover: true,
+              draggable: true,
+              progress: undefined
+            })
+          }
+        }
```

Comment:
early return

Reply:
what do you mean?

Reply:
Yep, in the backend

Reply:
if else is more complex to read than early return, prefer something like:
- you return earlier
- keep the simpler part in the if
- avoid unnecessary indentation and brackets

```ts
if (userCreate.error) {
  toast.error('This user already exists!', {
    position: toast.POSITION.TOP_RIGHT,
    autoClose: 2000,
    hideProgressBar: false,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    progress: undefined
  })

  return;
}
  
const token = userCreate.token

login(token);

navigate('/');

toast.success('Logged in!', {
  position: toast.POSITION.TOP_RIGHT,
  autoClose: 2000,
  hideProgressBar: false,
  closeOnClick: true,
  pauseOnHover: true,
  draggable: true,
  progress: undefined
})
```

Reply:
In more complex cases it's way way better, like the one in this article: https://gomakethings.com/the-early-return-pattern-in-javascript

Reply:
Hm, didn't know this was a pattern. Thanks for this, will read more about it.